src/pyddx_classes.cpp

```diff
@@ -432,20 +432,41 @@ class State {
   }
 
   // Obtain the COSMO / PCM / LPB forces. The state is modified in-place
-  array_f_t solvation_force_terms() {
+  array_f_t solvation_force_terms(py::object py_elec_property) {
```

Comment:
I don't like that the input argument is something else depending on the model. The issue is that this makes the interface more cumbersome to use (I as a user need to know something about the internals to use it).

I would suggest to better extract the respective entity directly out of a `py::dict` (e.g. use the key `"e"` for electric field, `"g"` for the gradient, error if key not around) and then expect the user to pass a matching dict. Let me know if you feel that goes outside of your comfort zone regarding the pybind11 stuff and I can do it.

Reply:
I will merge as is, but something to keep in mind.